core/dbt/parser/schemas.py

```diff
@@ -381,8 +386,16 @@ def _generate_test_name(self, target, test_type, test_args):
     def get_path(self, *parts):
         return '.'.join(str(s) for s in parts)
 
+    def _parse_identifier(self, identifier):
+        if identifier.startswith(self.quote_character) and \
+           identifier.endswith(self.quote_character):
+            return True, identifier[1:-1]
+        else:
+            return False, identifier
```

Comment:
if i have any doubt about this branch, it's this fn. the idea of it is fine, but i can see it breaking easily...

do we trim whitespace on identifiers specified in schema.yml?

Reply:
really good question! I think there's some amount of _caveat emptor_ that needs to happen here. I'm super happy to trim whitespace (this is a good idea) but i imagine that there are always going to be intractable edge cases with an approach like this. If your database starts and ends with quotation marks, then you have bigger problems than dbt's sources not working i think.

I'd agree with you that this is that part of the PR that i feel the least comfortable with. Happy to explore other ways of doing this, or possibly talk about to avoid string hackery all together. I think this functionality results in the nicest UX for users, but i could equivalently imagine:

```
sources:
  - name: snowplow
    database: raw
    quoting:
      database: true
    ....
```

I think that's doable, but i imagine it's unpleasant both for dbt devs and for users. Can think about this a little more for sure